Add --custom-function parameter to check

umutphp opened the pull request:
This pull request is the clone of [the pull request](https://github.com/JakubOnderka/PHP-Var-Dump-Check/pull/32) for the previous repository ([JakubOnderka/PHP-Var-Dump-Check](https://github.com/JakubOnderka/PHP-Var-Dump-Check)). 

Closes #5

Commit:
Add --custom-function parameter to add custom functions to the list of functions to be checked

Commit:
Add test for custom function parameter and add .idea folder to gitignore

umutphp committed:
Revert the .idea folder

umutphp committed:
Implement the CR "Please use ` instead of " to match other parameters description"

umutphp committed:
Empty is used instead of count($customFunctions) > 0

umutphp committed:
Merge branch 'master' into master

grogy committed:
Changed package name - new organization

grogy committed:
Fixed Travis CI configuration

grogy committed:
Added replace to composer for previous package

grogy committed:
Added run CI with PHP 7.4

grogy committed:
Fixed vendor names in readme for install

grogy committed:
Created version 0.4

reedy committed:
Create .gitattributes

grogy committed:
Added .gitattributes to changelog

peter279k committed:
Replace array syntax with short array syntax

grogy committed:
Added PHP array syntax to changelog

grogy committed:
Added cleaning changes to changelog

jrfnl committed:
PHP 7.4: fix compatibility issue

Array access using curly braces has been deprecated in PHP 7.4.

Ref: https://www.php.net/manual/en/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace

jrfnl committed:
Composer: allow installation of more recent Parallel Lint

Using the caret is the recommended manner of setting a version requirement.

grogy committed:
Added internal change to changelog

grogy committed:
Added fix for PHP 7.4 to changelog

grogy commented on src/Settings.php:
```diff
@@ -138,7 +138,7 @@ public static function parseArguments(array $arguments)
 
         // Any custom function is given
-        if (count($customFunctions) > 0) {
+        if (!empty($customFunctions)) {
```
Please, why here is used `empty`?

grogy commented on README.md:
```diff
@@ -48,7 +48,7 @@ Options for run
 - `--symfony` - check dump: `dump`, `VarDumper::dump`, `VarDumper::setHandler`
 - `--laravel` - check dump: `dd`, `dump`
-- `--custom-function` - comma separated custom function name(s) to check like "pre_echo"
+- `--custom-function` - comma separated custom function name(s) to check like `pre_echo`
```
This commit can be squash with implementation

grogy commented on .gitignore:
```diff
@@ -1,3 +1,2 @@
 /vendor/
 /composer.lock
-.idea
```
This commit can be squashed with implementing commit

grogy commented on tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php:
```diff
@@ -0,0 +1,65 @@
+<?php
```
Please, why isn't this commit included in implementation? I mean that will be better for people who read Git history :)

grogy commented on tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php:
```diff
@@ -0,0 +1,65 @@
+
+class CustomFunctionTest extends PHPUnit_Framework_TestCase
+{
+    public function __construct()
```
Empty construction we can remove, what do you mean?

grogy commented on tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php:
```diff
@@ -0,0 +1,65 @@
+
+    }
+
+
```
What about add test for none function founded? (I mean when code does not includes `functionName1`) It coverage all cases

grogy commented on src/Settings.php:
```diff
@@ -70,8 +70,9 @@ class Settings
     {
-        $arguments = new ArrayIterator(array_slice($arguments, 1));
-        $setting = new self;
+        $arguments       = new ArrayIterator(array_slice($arguments, 1));
```
Please do not horizontal align code, it is not used more in the codebase

grogy reviewed (commented):
Perfect! Thank you for your PR. It looks great.

Here are some comments for improve code and repository history.
1. Can you help with squash commits? We can call together (like whereby.com) and I can show you how do it. 
2. Can you rebase it to current master?
3. We can add changelog line, it is perfect new feature.

umutphp commented:
Hi @grogy ,

Many thanks for the review and feedbacks. This was a very old PR coming from the old repo :). I will fix the problems and make the PR ready for merge asap.

umutphp committed:
Merge remote-tracking branch 'upstream/master'

umutphp committed:
Add test for none function founded

Caused by the review message in the following link.
https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470564342

umutphp committed:
Add test for none function founded

Caused by the review message in the following link.
https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470564342

umutphp committed:
Remove horizontal align

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470566935

umutphp committed:
Remove horizontal align

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470566935

umutphp committed:
Remove unneccessary empty check

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470562392

umutphp committed:
Remove unneccessary empty check

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470562392

umutphp committed:
Add the Symfony Vardumper dd() helper method introduced in 4.1

umutphp commented:
I think I need to close this PR and come with a clean PR.
